From b454a60bf154332bd5961880700b7d08aec98995 Mon Sep 17 00:00:00 2001
Message-Id: <b454a60bf154332bd5961880700b7d08aec98995.1508856243.git.jan.steffens@gmail.com>
From: "Jan Alexander Steffens (heftig)" <jan.steffens@gmail.com>
Date: Tue, 24 Oct 2017 16:29:49 +0200
Subject: [PATCH] meson: Fix GDB scripts install_dir for *nix

Disable installation on Windows for now as this would use a colon in the
directory name.

https://bugzilla.gnome.org/show_bug.cgi?id=788772
---
 glib/meson.build    | 5 +++--
 gobject/meson.build | 5 +++--
 2 files changed, 6 insertions(+), 4 deletions(-)

diff --git a/glib/meson.build b/glib/meson.build
index 1da0bcd3cc601b9e..33ece2d0221c1850 100644
--- a/glib/meson.build
+++ b/glib/meson.build
@@ -283,8 +283,9 @@ configure_file(
   input: 'libglib-gdb.py.in',
   output: 'libglib-2.0.so.@0@-gdb.py'.format(library_version),
   configuration: gdb_conf,
-  install: true,
-  install_dir: join_paths(get_option('datadir'), 'gdb/auto-load' + get_option('libdir'))
+  # FIXME: Figure out how to install this on Windows
+  install: host_system != 'windows',
+  install_dir: join_paths(glib_datadir, 'gdb', 'auto-load', './' + glib_libdir)
 )
 
 if enable_systemtap
diff --git a/gobject/meson.build b/gobject/meson.build
index 5a8289e6f054818e..0c8c0cb26f053d1c 100644
--- a/gobject/meson.build
+++ b/gobject/meson.build
@@ -100,8 +100,9 @@ configure_file(
   input: 'libgobject-gdb.py.in',
   output: 'libgobject-2.0.so.@0@-gdb.py'.format(library_version),
   configuration: gdb_conf,
-  install: true,
-  install_dir: join_paths(get_option('datadir'), 'gdb/auto-load/' + get_option('libdir'))
+  # FIXME: Figure out how to install this on Windows
+  install: host_system != 'windows',
+  install_dir: join_paths(glib_datadir, 'gdb', 'auto-load', './' + glib_libdir)
 )
 
 if enable_systemtap
-- 
2.14.2

